pkg/arvo/sys/hoon.hoon

```diff
@@ -1773,23 +1766,23 @@
     =+  b=$(a r.a)
     :-  p.b
     ?:  |(?=(~ q.b) (mor n.a n.q.b))
-      [n.a l.a q.b]
-    [n.q.b [n.a l.a l.q.b] r.q.b]
+      a(r q.b)
+    a(n n.q.b, l a(r l.q.b), r r.q.b)
   ::
-  ++  nip                                               ::  remove root
+  ++  nip                                               ::  removes root
     |-  ^+  a
     ?~  a  ~
     ?~  l.a  r.a
     ?~  r.a  l.a
     ?:  (mor n.l.a n.r.a)
-      [n.l.a l.l.a $(l.a r.l.a)]
-    [n.r.a $(r.a l.r.a) r.r.a]
+      l.a(r $(l.a r.l.a))
+    r.a(l $(r.a l.r.a))
   ::
-  ++  nap                                               ::  removes head
+  ++  nap                                               ::  removes root
     ?>  ?=(^ a)
     ?:  =(~ l.a)  r.a
     =+  b=get(a l.a)
-    bal(a ^+(a [p.b q.b r.a]))
+    bal(a ^+(a a(n p.b, l q.b)))
```

Comment:
I wonder if you could shorten this:
`bal(n.a p.b, l.a q.b)`

Reply:
Yeah, this seems to work.